### PR TITLE
feat(audit-workspace): 10-bin classifier finding schema (slice 8b, #203)

### LINKED ISSUE #203: 10-bin classifier output JSON schema (compliance_risk + expected_token_efficiency_rank + cache_strategy) (slice 8b)
## What to build

10-bin classifier output JSON schema. Output of every classifier finding is validated against this schema before being included in the dry-run report. Schema covers: destination bin (one of `Delete | Locality 0 | Locality 1 | Locality 2 | Locality 3a | Locality 3b | Locality 3c | Locality 3d | Locality 3e | Locality 4`), `rationale`, `compliance_risk` (`deterministic` | `agent-dependent`), `expected_token_efficiency_rank` (numeric), `cache_strategy` (from slice 8a), and `suggested_artifact_path`.

Per Decisions Q4, Q5, Q11.

Source plan: `docs/ideas/audit-workspace-improve-flow.md` Phase 4 (10-bin classifier section).

## Acceptance criteria

- [ ] JSON schema file lives at `.github/skills/audit-knowledgebase-workspace/schema/finding.schema.json` (or equivalent governed-schema path)
- [ ] Schema validates all 10 destination bins
- [ ] `compliance_risk` enum: `deterministic` (Locality 0 + Locality 3a/3b/3c/3d) or `agent-dependent` (Locality 1, Locality 2, Locality 3e, Locality 4 mechanism-A)
- [ ] `expected_token_efficiency_rank` is required; computation rule documented in ADR-028 reference
- [ ] `cache_strategy` field present (consumed from slice 8a output)
- [ ] Test fixtures: 10 example findings (one per destination); 2 example findings per `compliance_risk` value; 1 invalid example (missing required field) → schema rejection
- [ ] Optional: register the schema in `schema/` index per repo convention if applicable
- [ ] `python3 -m pytest tests/kb/test_audit_workspace_finding_schema.py` passes

## Blocked by

- #202 (slice 8a cache must exist for cache_strategy field to be populated)

### PR DESCRIPTION
**Pull request by wryenmeek**
Implements #203 — JSON Schema for the 10-bin classifier output (slice 8b).

## Acceptance status

- [x] Schema file at `.github/skills/audit-knowledgebase-workspace/schema/finding.schema.json`
- [x] Validates all 10 destination bins
- [x] `compliance_risk` enum: `deterministic` | `agent-dependent`
- [x] `expected_token_efficiency_rank` required, integer per Phase 4 token-efficiency rule
- [x] `cache_strategy` field present and compatible with slice 8a `mtime_first_para` plus future values
- [x] Test fixtures: 10 destination examples + 4 compliance_risk examples + invalid examples
- [x] Repo-relative path hardening for `source_file` and `suggested_artifact_path`
- [x] pytest passing

## Validation

- `python3 -m pytest tests/kb/test_audit_workspace_finding_schema.py -v` → 8 passed, 29 subtests passed
- `python3 -m pytest tests/kb/test_audit_workspace_cache.py tests/kb/test_context_md_freshness.py tests/kb/test_audit_workspace_finding_schema.py -q` → 21 passed, 38 subtests passed
- CI check-runs on the PR head are success/skipped only as of the latest REST poll.

## Cross-functional review

Dispatched code, test, security, and documentation reviewers. One P2 security finding around unsafe repo-relative path values was remediated; final security re-review reported no blocking findings. Gemini review comments about rank and path edge-case test coverage were addressed in the amended commit and replied to via REST.

Closes #203.

### CODE REVIEW

File: tests/kb/test_audit_workspace_finding_schema.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        finding = self._finding(compliance_risk="maybe-deterministic")
+
+        with self.assertRaisesRegex(SchemaValidationError, "compliance_risk"):
+            self._validate_with_schema(finding, self.schema)
+
+    def test_schema_rejects_non_numeric_expected_token_efficiency_rank(self) -> None:
+        finding = self._finding(expected_token_efficiency_rank="3")
+
+        with self.assertRaisesRegex(SchemaValidationError, "expected_token_efficiency_rank"):
+            self._validate_with_schema(finding, self.schema)
```
[the remark this answers is not in the document]
Comment (wryenmeek) [reply tests/kb/test_audit_workspace_finding_schema.py]: Addressed in 54f5bcf: expanded expected_token_efficiency_rank rejection coverage for string, float, and boolean values.

File: tests/kb/test_audit_workspace_finding_schema.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            ("source_file", "/etc/passwd"),
+            ("source_file", "file:///AGENTS.md"),
+            ("suggested_artifact_path", "../../outside"),
+            ("suggested_artifact_path", "docs//double-slash.md"),
+            ("suggested_artifact_path", "docs/run;rm.md"),
+            ("suggested_artifact_path", "docs/has space.md"),
+            ("source_file", "docs/good.md\n"),
+            ("suggested_artifact_path", "docs/good.md\n"),
+            ("suggested_artifact_path", "docs/good.md\t"),
+        )
```
[the remark this answers is not in the document]
Comment (wryenmeek) [reply tests/kb/test_audit_workspace_finding_schema.py:162]: Addressed in 54f5bcf: added unsafe path regression cases for backslashes, middle colons, shell metacharacters, trailing slashes, and single-dot segments.
